packages/typespec-ts/test/integration/generated/routes/src/index.d.ts

```diff
@@ -453,12 +498,18 @@ export declare interface QueryParametersQueryContinuationStandardRecord204Respon
 
 export declare type QueryParametersQueryContinuationStandardRecordParameters = QueryParametersQueryContinuationStandardRecordQueryParam & RequestParameters;
 
+export declare interface QueryParametersQueryContinuationStandardRecordParamQueryParam {
+    value: Record<string, number>;
+    explode: false;
+    style: "form";
+}
+
 export declare interface QueryParametersQueryContinuationStandardRecordQueryParam {
     queryParameters: QueryParametersQueryContinuationStandardRecordQueryParamProperties;
 }
 
 export declare interface QueryParametersQueryContinuationStandardRecordQueryParamProperties {
-    param: Record<string, number>;
+    param: QueryParametersQueryContinuationStandardRecordParamQueryParam;
```

Comment:
is there a case where value type is array of objects?

Reply:
for complicated nested cases e.g objected nested array or array nested objects i don't think we have agreement on how to serialize them correctly. our code is not ready either.

i think this would be a cross-language issue and prefer to defer any discussion for them for now considering no real usage.

Reply:
create an issue for complicated nested cases.

Reply:
https://github.com/Azure/autorest.typescript/issues/2906